src/ReshowcaseUi.re

```diff
@@ -1097,64 +1071,53 @@ module App = {
             ->(Option.map(demoUnit => <DemoUnit demoUnit />))
             ->(Option.getWithDefault("Demo not found"->React.string))}
          </div>;
-       | Demo(queryString) =>
-         <>
-           <DemoListSidebar
-             demos
-             urlSearchParams
-             isCategoriesCollapsedByDefault
-             onToggleCollapsedCategoriesByDefault
+       | Demo(demoName) =>
+         let demoUnit =
+           Demos.findDemo(urlSearchParams, demoName, demos)
+           ->(
+               Option.map(demoUnit =>
+                 <DemoUnit demoUnit key={url.search} ?sidebarElem />
+               )
+             )
+           ->(Option.getWithDefault("Demo not found"->React.string));
+         <div name="Content" style=Styles.right>
+           <TopPanel
+             isSidebarHidden={!showRightSidebar}
+             responsiveMode
+             onRightSidebarToggle={() => {
+               toggleShowRightSidebar(_ => !showRightSidebar);
+               switch (loadedIframeWindow) {
+               | Some(window) when !showRightSidebar =>
+                 Window.postMessage(window, RightSidebarDisplayed)
```

Comment:
I am not sure why it's done via "signalling", but decided to keep it this way for now to limit the scope of this PR